lib/install/client/install.go

```diff
@@ -146,16 +146,19 @@ type InstallerStrategy struct {
 	ConnectTimeout time.Duration
 }
 
-// isServiceFailed returns an error if the service has failed.
-func isServiceFailed(serviceName string) func() error {
+// shouldReconnectService returns a function that determines whether the client should continue
+// reconnecting to the service given with serviceName
+func shouldReconnectService(serviceName string) func() error {
 	return func() error {
-		err := service.IsFailed(serviceName)
+		err := service.IsStatus(serviceName,
+			systemservice.ServiceStatusFailed)
```

Comment:
Removed the `unknown` state from the list as it is really uncertain whether selecting on this active state is worth it or would make the selector (`shouldReconnectService`) racy in any way.